tests/integ/test_model_package.py

```diff
@@ -0,0 +1,87 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import os
+from sagemaker.model_card.schema_constraints import ModelApprovalStatusEnum
+from sagemaker.utils import unique_name_from_base
+from sagemaker.model import ModelPackage
+from tests.integ import DATA_DIR
+from sagemaker.xgboost import XGBoostModel
+
+_XGBOOST_PATH = os.path.join(DATA_DIR, "xgboost_abalone")
+
+
+def test_add_remove_model_groups_in_collection_success(sagemaker_session):
+
+    model_group_name = unique_name_from_base("test-model-group")
+
+    sagemaker_session.sagemaker_client.create_model_package_group(
+        ModelPackageGroupName=model_group_name
+    )
+
+    xgb_model_data_s3 = sagemaker_session.upload_data(
+        path=os.path.join(_XGBOOST_PATH, "xgb_model.tar.gz"),
+        key_prefix="integ-test-data/xgboost/model",
+    )
+    model = XGBoostModel(
+        model_data=xgb_model_data_s3, framework_version="1.3-1", sagemaker_session=sagemaker_session
+    )
+    image_uri = model.serving_image_uri(
+        region_name=sagemaker_session.boto_session.region_name,
+        instance_type="ml.p2.xlarge",
+    )
+    create_model_package_input_dict = {
+        "ModelPackageGroupName": model_group_name,
+        "ModelPackageDescription": "Test model package registered for integ test",
+        "ModelApprovalStatus": ModelApprovalStatusEnum.PENDING_MANUAL_APPROVAL,
+        "InferenceSpecification": {
+            "Containers": [{"Image": image_uri}],
+            "SupportedContentTypes": ["text/csv"],
+            "SupportedResponseMIMETypes": ["text/csv"],
+            "SupportedRealtimeInferenceInstanceTypes": ["ml.m5.large"],
+            "SupportedTransformInstanceTypes": ["ml.m5.large"],
+        },
+    }
+
+    create_model_package_resp = sagemaker_session.sagemaker_client.create_model_package(
+        **create_model_package_input_dict
+    )
+    model_package_arn = create_model_package_resp["ModelPackageArn"]
+    desc_model_package = sagemaker_session.sagemaker_client.describe_model_package(
+        ModelPackageName=model_package_arn
+    )
+    model_package_arn = desc_model_package["ModelPackageArn"]
+    approval_status = desc_model_package["ModelApprovalStatus"]
+
+    assert approval_status == ModelApprovalStatusEnum.PENDING_MANUAL_APPROVAL
+
+    model_package = ModelPackage(
+        model_package_arn=model_package_arn,
+        sagemaker_session=sagemaker_session,
+    )
```

Comment:
Instead of calling api sdk we should call register() function on XGBoostModel itself and this should work out of box. You might need to pass content_types and response_types for now to register model step.